toolchain/check/handle_class.cpp

```diff
@@ -653,10 +646,11 @@ static auto CheckCompleteClassType(Context& context, Parse::NodeId node_id,
 
   bool defining_vtable_ptr = class_info.is_dynamic;
   if (class_info.base_id.is_valid()) {
-    auto base_info = context.insts().GetAs<SemIR::BaseDecl>(class_info.base_id);
+    auto base_type_id =
+        class_info.GetBaseType(context.sem_ir(), SemIR::SpecificId::Invalid);
```

Comment:
Looks like this could be refactored to be closer to what you're doing elsewhere:

```suggestion
  if (auto base_type_id =
          class_info.GetBaseType(context.sem_ir(), SemIR::SpecificId::Invalid); base_type_id.is_valid()) {